libraries/stdlib/jdk7/src/kotlin/io/PathUtils.kt

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER", "RedundantVisibilityModifier")
+@file:JvmMultifileClass
+@file:JvmName("PathsKt")
+@file:kotlin.jvm.JvmPackageName("kotlin.io.jdk7")
+
+package kotlin.io
+
+import java.io.IOException
+import java.nio.file.*
+import java.nio.file.FileAlreadyExistsException
+import java.nio.file.NoSuchFileException
+
+/**
+ * Returns the extension of this path (not including the dot), or an empty string if it doesn't have one.
+ */
+@SinceKotlin("1.4")
+@ExperimentalStdlibApi
+public val Path.extension: String
+    get() = fileName.toString().substringAfterLast('.', "")
+
+/**
+ * Returns this path as a [String] using the invariant separator '/' to
+ * separate the names in the name sequence.
+ */
+@SinceKotlin("1.4")
+@ExperimentalStdlibApi
+public val Path.invariantSeparatorsPath: String
+    get() {
+        val separator = fileSystem.separator
+        return if (separator != "/") toString().replace(separator, "/") else toString()
+    }
+
+/**
+ * Returns this path's [fileName][Path.getFileName] without an extension.
+ */
+@SinceKotlin("1.4")
+@ExperimentalStdlibApi
+public val Path.nameWithoutExtension: String
+    get() = fileName.toString().substringBeforeLast(".")
```

Comment:
`fileName` could be `null`. I doubt that we want to have `nameWithoutExtension` return "null" string in this case.